index_io.go

```diff
@@ -12,30 +11,20 @@ import (
 	"unsafe"
 )
 
-// WriteIndex writes an index to a file.
-func WriteIndex(idx Index, filename string) error {
-	cfname := C.CString(filename)
-	defer C.free(unsafe.Pointer(cfname))
-	if c := C.faiss_write_index_fname(idx.cPtr(), cfname); c != 0 {
-		return getLastError()
-	}
-	return nil
-}
-
 func WriteIndexIntoBuffer(idx Index) ([]byte, error) {
 	runtime.LockOSThread()
 	defer runtime.UnlockOSThread()
 
 	// the values to be returned by the faiss APIs
-	tempBuf := (*C.uchar)(C.malloc(C.size_t(0)))
+	var tempBuf *C.uchar = nil
 	bufSize := C.size_t(0)
 
 	if c := C.faiss_write_index_buf(
 		idx.cPtr(),
 		&bufSize,
 		&tempBuf,
 	); c != 0 {
-		C.free(unsafe.Pointer(tempBuf))
+		C.faiss_free_buf(tempBuf)
```

Comment:
Would you still need to free in case of err path?

Reply:
it was already there so i assumed that a free is required